src/routes/safe/components/Apps/components/SafeAppCard/SafeAppCard.tsx

```diff
@@ -0,0 +1,238 @@
+import { copyToClipboard } from 'src/utils/clipboard'
+import { getShortName } from 'src/config'
+import { SafeAppDescriptionSK, SafeAppLogoSK, SafeAppTitleSK } from './SafeAppSkeleton'
+
+type SafeAppCardSize = 'md' | 'lg'
+
+type SafeAppCardProps = {
+  safeApp: SafeApp
+  size: SafeAppCardSize
+  togglePin: (app: SafeApp) => void
+  isPinned?: boolean
+  isCustomSafeApp?: boolean
+  onRemove?: (app: SafeApp) => void
+}
+
+const SafeAppCard = ({
+  safeApp,
+  size,
+  togglePin,
+  isPinned,
+  isCustomSafeApp,
+  onRemove,
+}: SafeAppCardProps): React.ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const dispatch = useDispatch()
+
+  const safeAddress = extractSafeAddress()
+  const appsPath = generateSafeRoute(SAFE_ROUTES.APPS, {
+    shortName: getShortName(),
+    safeAddress,
+  })
+  const openSafeAppLink = `${appsPath}?appUrl=${encodeURI(safeApp.url)}`
+
+  const shareSafeApp = () => {
+    const shareSafeAppUrl = getShareSafeAppUrl(safeApp.url, chainId)
+    copyToClipboard(shareSafeAppUrl)
+    dispatch(enqueueSnackbar(enhanceSnackbarForAction(NOTIFICATIONS.SHARE_SAFE_APP_URL_COPIED)))
+  }
+
+  const isSafeAppLoading = safeApp.fetchStatus === FETCH_STATUS.LOADING
+
+  if (isSafeAppLoading) {
+    return (
+      <SafeAppContainer size={size} layout initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }}>
+        <StyledAppCard size={size}>
+          <LogoContainer size={size}>
+            <SafeAppLogoSK size={size} />
+          </LogoContainer>
+          <DescriptionContainer size={size}>
+            <SafeAppTitleSK />
+            <SafeAppDescriptionSK />
+            <SafeAppDescriptionSK />
+          </DescriptionContainer>
+        </StyledAppCard>
+      </SafeAppContainer>
+    )
+  }
+
+  return (
+    <SafeAppContainer size={size} layout initial={{ opacity: 0 }} animate={{ opacity: 1 }} exit={{ opacity: 0 }}>
+      <StyledLink to={openSafeAppLink} aria-label={`open ${safeApp.name} Safe App`}>
+        <StyledAppCard size={size}>
+          {/* Safe App Logo */}
+          <LogoContainer size={size}>
+            <SafeAppLogo
+              size={size}
+              src={safeApp.iconUrl}
+              alt={`${safeApp.name || 'Safe App'} Logo`}
+              onError={setSafeAppLogoFallback}
+            />
+          </LogoContainer>
+
+          {/* Safe App Description */}
+          <DescriptionContainer size={size}>
+            <SafeAppTitle size="xs">{safeApp.name}</SafeAppTitle>
+            <SafeAppDescription size="lg">{safeApp.description}</SafeAppDescription>
+          </DescriptionContainer>
+
+          {/* Safe App Actions */}
+          <ActionsContainer onClick={(e) => e.preventDefault()}>
+            {/* Share Safe App button */}
+            <IconBtn onClick={shareSafeApp} aria-label={`copy ${safeApp.name} Safe App share link to clipboard`}>
+              <Icon size="md" type="share" tooltip="copy share link to clipboard" />
```

Comment:
```suggestion
            <IconBtn onClick={shareSafeApp} aria-label={`Copy shareable ${safeApp.name} Safe App link to clipboard`}>
              <Icon size="md" type="share" tooltip="Copy shareable link to clipboard" />
```

Reply:
“Copy share link”
“Unpin from the Safe Apps”
“Remove Custom Safe App”
Lets keep it short

Reply:
`Unpin Safe App` :)